app/app.go

```diff
@@ -173,6 +179,19 @@ func NewApp(
 	logger log.Logger, db dbm.DB, tio io.Writer, loadLatest bool, invCheckPeriod uint, skipUpgradeHeights map[int64]bool,
 	homePath string, appOpts servertypes.AppOptions, options ...func(*bam.BaseApp),
 ) *AkashApp {
+	// find out the genesis time, to be used later in inflation calculation
+	if v := appOpts.Get("GenesisTime"); v != nil {
+		// in tests, GenesisTime is supplied using appOpts
+		genTime, ok := v.(time.Time)
+		if !ok {
+			panic("expected GenesisTime to be a Time value")
+		}
+		inflationtypes.GenesisTime = genTime
+	} else if genDoc, err := tmtypes.GenesisDocFromFile(filepath.Join(homePath, "config/genesis.json")); err != nil {
+		panic(err)
+	} else {
+		inflationtypes.GenesisTime = genDoc.GenesisTime
+	}
```

Comment:
Put this in a helper function that returns the genesis time.

Please don't use global variables for state.  Find another way to pass the parameters into the inflation calculation.

Reply:
Done